diff --git a/Makefile b/Makefile
index 35cf2f5..f0bccf9 100644
--- a/Makefile
+++ b/Makefile
@@ -30,12 +30,12 @@
 # line when debugging.
 OMIT_FRAME_PTR = -fomit-frame-pointer
 
-LUABIN = lua
-LUAPKG = lua5.2
-CFLAGS = `pkg-config $(LUAPKG) --cflags` -fPIC -O3 -Wall $(OMIT_FRAME_PTR)
-LFLAGS = -shared
+#LUABIN = lua
+#LUAPKG = lua5.2
+#CFLAGS = `pkg-config $(LUAPKG) --cflags` -fPIC -O3 -Wall $(OMIT_FRAME_PTR)
+#LFLAGS = -shared
 
-INSTALL_PATH = `$(LUABIN) -e'                           \
+#INSTALL_PATH = `$(LUABIN) -e'                           \
     for dir in package.cpath:gmatch("(/[^?;]+)?") do    \
         io.write(dir)                                   \
         os.exit(0)                                      \
@@ -50,7 +50,15 @@ INSTALL_PATH = `$(LUABIN) -e'                           \
 #CFLAGS = -I/usr/local/include/ -fPIC -O3 -Wall $(OMIT_FRAME_PTR)
 #LFLAGS = -shared
 #INSTALL_PATH = /usr/local/lib/lua/5.2/
+LUA_VERSION=5.1
+PREFIX ?= /usr/local
+LUA_INCLUDE_DIR ?= $(PREFIX)/include
+LUA_LIB_DIR ?= $(PREFIX)/lib/lua/$(LUA_VERSION)
+CFLAGS ?= -I$(LUA_INCLUDE_DIR) -fPIC -O3 -Wall
+override CFLAGS += $(OMIT_FRAME_PTR)
+LFLAGS += -shared
 
+INSTALL ?= install
 
 all: iconv.so
 
@@ -62,7 +70,9 @@ iconv.so: iconv.lo
 
 install: iconv.so
 	make test
-	install -D -s iconv.so $(DESTDIR)/$(INSTALL_PATH)/iconv.so
+	$(INSTALL) -d $(DESTDIR)/$(LUA_LIB_DIR)
+	$(INSTALL) iconv.so $(DESTDIR)/$(LUA_LIB_DIR)
+	#install -D -s iconv.so $(DESTDIR)/$(INSTALL_PATH)/iconv.so
 
 clean:
 	$(RM) iconv.so iconv.lo

